### PR TITLE
add maxFiles to GangaDatasetSplitter

### LINKED ISSUE #1094: Add a maxFiles option to GangaDatasetSplitter
You should be able to specify the maximum number of files to split over in `GangaDatasetSplitter`, c.f. `maxFiles` in `SplitByFiles`.

### PR DESCRIPTION
**Pull request by mesmith75**
Fixes #1094 . Also added in missing import to Job.

### CODE REVIEW

File: python/Ganga/Lib/Splitters/GangaDatasetSplitter.py [hunk 1]
```diff
@@ -18,6 +18,10 @@ class GangaDatasetSplitter(ISplitter):
     _name = "GangaDatasetSplitter"
     _schema = Schema(Version(1, 0), {
         'files_per_subjob': SimpleItem(defvalue=5, doc='the number of files per subjob', typelist=[int]),
+
+        'maxFiles': SimpleItem(defvalue=None,
+                               doc='Maximum number of files to use in a masterjob (None = all files)',
+                               typelist=['int', 'type(None)']),
```
Comment (rob-c) [inline python/Ganga/Lib/Splitters/GangaDatasetSplitter.py]: I think `typelist=[int,None]` may be better

File: python/Ganga/Lib/Splitters/GangaDatasetSplitter.py [hunk 2]
```diff
@@ -48,7 +52,10 @@ def split(self, job):
         # split based on all the sub files
         fid = 0
         subjobs = []
-        while fid < len(full_list):
+        filesToRun = len(full_list)
+        if self.maxFiles and not self.maxFiles == -1:
```
Comment (rob-c) [inline python/Ganga/Lib/Splitters/GangaDatasetSplitter.py]: I like the `maxFiles=-1` should that be the default in the schema?

File: python/Ganga/Lib/Splitters/GangaDatasetSplitter.py (hunk 1 above)
Comment (mesmith75) [inline python/Ganga/Lib/Splitters/GangaDatasetSplitter.py]: Yes, I'll change that.

File: python/Ganga/Lib/Splitters/GangaDatasetSplitter.py (hunk 2 above)
Comment (mesmith75) [inline python/Ganga/Lib/Splitters/GangaDatasetSplitter.py]: Yes

**Comment by mesmith75** [comment]
Can I merge this?

**Pull request merged**